pkg/service/service.go

```diff
@@ -861,10 +861,10 @@ func (s *Service) upsertService(params *lb.SVC) (bool, lb.ID, error) {
 		return false, lb.ID(0), err
 	}
 
-	// Update managed neighbor entries of the LB
-	if option.Config.LoadBalancerOnly {
-		s.upsertBackendNeighbors(newBackends, obsoleteBackends)
-	}
+	// Update managed neighbor entries of the LB, this is needed so that
+	// neighbor entries for the backends are always up to date if they
+	// reside in the same L2. In particular XDP cannot resolve on-demand.
+	s.upsertBackendNeighbors(newBackends, obsoleteBackends)
```

Comment:
UpsertBackendNeighbors deletes the neighbor entries of nodes where the obsolete backends are running. Could this cause a connection disruption if an active backend, selected by another service, is running on those nodes whose neighbor entries were deleted?

Reply:
From a Cilium PoV these are refcounted internally in the agent's neighbor handling code, so they are only deleted once the last active one goes away.

Reply:
Right, I missed that